### PR TITLE
[ETCM-126] split JsonRpcControllerSpec

### PR DESCRIPTION
**Pull request by lemastero**
# Description

JsonRpcControllerSpec file contained around 2000 lines of code. This is hard to maintain. Context for this PR: https://github.com/input-output-hk/mantis/pull/726#discussion_r503406911

# Proposed Solution

Split it into tests related to:
- `JsonRpcControllerPersonalSpec` personal_XYZ
- `JsonRpcControllerEthTransactionSpec` related to eth transaction requests
- `JsonRpcControllerEthSpec` other eth requests
- `JsonRpcControllerSpec` other requests

Before split there were utility function extracted that simplify creating requests and creating JsonRpcService: `newJsonRpcController`, `newJsonRpcRequest` in https://github.com/input-output-hk/mantis/pull/734/commits/53283b73265c070be4b22062a3bbad7fab577a04

# Important Changes Introduced

# Testing

### CODE REVIEW

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    xs.lift(idx)
+      .map(encodeSignedTrx)
+
+  def encodeSignedTrx(x: SignedTransaction) =
+    encodeAsHex(RawTransactionCodec.asRawTransaction(x))
+
+  val version = Config.clientVersion
+  val blockGenerator = mock[EthashBlockGenerator]
+
+  override implicit lazy val system = ActorSystem("JsonRpcControllerSpec_System")
```
Comment (mmrozek) [inline src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala:37]: Do we need to create ActorSystem for each test?

**Review by mmrozek — APPROVED** [review]
Nice job. One doubt only

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala (hunk 1 above)
Comment (kapke) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala:37]: This.
I'd say it's preferrable to use akka's `TestKit` on suite level and use there `WithActorSystemShutdown` trait.
Benefits are as follows:
1. Creating and shutting down systems is quite expensive operation.
2. Actors of any kind are created per spec anyways so lack of isolation is not an issue I think
3. In other project we had real problems with tests due to leaking running actor systems

**Review by kapke — COMMENTED** [review]
Overall looks good to me!

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala (hunk 1 above)
Comment (lemastero) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala:37]: How about merging this PR and fix the unnecessary spawning actor system in a separate one.

There is a pretty complex wiring using cake pattern (actor system that comes from `ActorSystemBuilder` (via `StdTestConsensusBuilder` via `ScenarioSetup` via `EphemBlockchainTestSetup` and ActorSystemBilder is required in `StdLedgerBuilder` via self type). I need some time to figure out how to do this.

Comment (kapke) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala:37]: Sounds good to me. And such task/PR could be extended to making sure actor system is properly closed in all tests.

Comment (lemastero) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala:37]: PR for this change: https://github.com/input-output-hk/mantis/pull/751

**Pull request merged**